tests/bot/cogs/moderation/test_utils.py

```diff
@@ -0,0 +1,411 @@
+import textwrap
+import unittest
+from datetime import datetime
+from unittest.mock import AsyncMock, MagicMock, call, patch
+
+from discord import Embed, Forbidden, HTTPException, NotFound
+
+from bot.api import ResponseCodeError
+from bot.cogs.moderation import utils
+from bot.constants import Colours, Icons
+from tests.helpers import MockBot, MockContext, MockMember, MockUser
+
+
+class ModerationUtilsTests(unittest.IsolatedAsyncioTestCase):
+    """Tests Moderation utils."""
+
+    def setUp(self):
+        self.bot = MockBot()
+        self.member = MockMember(id=1234)
+        self.user = MockUser(id=1234)
+        self.ctx = MockContext(bot=self.bot, author=self.member)
+
+    async def test_user_get_active_infraction(self):
+        """
+        Should request the API for active infractions and return infraction if the user has one or `None` otherwise.
+
+        A message should be sent to the context indicating a user already has an infraction, if that's the case.
+        """
+        test_cases = [
+            {
+                "get_return_value": [],
+                "expected_output": None,
+                "infraction_nr": None,
+                "send_msg": True
+            },
+            {
+                "get_return_value": [{"id": 123987}],
+                "expected_output": {"id": 123987},
+                "infraction_nr": "123987",
+                "send_msg": False
+            },
+            {
+                "get_return_value": [{"id": 123987}],
+                "expected_output": {"id": 123987},
+                "infraction_nr": "123987",
+                "send_msg": True
+            }
+        ]
```

Comment:
Not asking you to change these, but wouldn't it be better to have a `namedtuple` or a similar construct to define the cases? The repeated subscription with string literals seems fragile and needlessly inconvenient.

Reply:
I added `namedtuple` for some test cases, but for others, this will be such mess and no one will understand, what thing is what, especially these what have embeds/many keys.